src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  sourceString.split(';').forEach((item) => {
+    const pairs = item.trim().split(':');
+    let key;
+    let value;
+
+    if (pairs.length === 2) {
+      key = pairs[0].trim();
+      value = pairs[1].trim();
+    }
+    obj[key] = value;
```

Comment:
The assignment `obj[key] = value;` should be inside the `if` block to prevent `undefined` keys from being added to the object when the `pairs.length` is not equal to 2.